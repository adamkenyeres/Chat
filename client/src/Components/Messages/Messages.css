.Messages-list {
    padding: 20px 0;
    max-width: 900px;
    height: 830px;
    width: 100%;
    margin: 0 auto;
    list-style: none;
    padding-left: 0;
    flex-grow: 1;
    overflow-y:auto;
  }
  .Messages-message {
    display: flex;
    margin-top: 10px;
  }
  .Message-text {
    padding: 10px;
    max-width: 400px;
    margin: 0;
    border-radius: 12px;
    background-color: cornflowerblue;
    color: white;
    display: inline-block;
  }