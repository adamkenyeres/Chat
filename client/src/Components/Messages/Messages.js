import React, { Component } from 'react';
import './Messages.css'
class Messages extends Component {
    render() {
        const { messages } = this.props;
        return (
            <ul className="Messages-list">
                {messages.map(m => this.renderMessage(m))}
            </ul>
        );
    }
    renderMessage(message) {
        const {user, text} = message;
        return (
            <li className="Messages-message">
                <div className="Message-content">
                    <div className = "message-user">
                        {user}
                    </div>
                    <div className="Message-text">
                        {text}
                    </div>
                </div>
            </li>
        );
    }
}

export default Messages;
