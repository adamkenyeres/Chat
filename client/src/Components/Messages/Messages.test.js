import React from 'react';
import { shallow } from 'enzyme';
import Messages from './Messages';


describe('Messages', () => {
    it('should render correctly with one message', () => {
        const messages = [ { text: 'Hi How are you?', user: 'Adam' }];
        const component = shallow(<Messages messages = {messages} />);
          
          expect(component).toMatchSnapshot();
        });

    it('should render correctly', () => {
        const messages = [ { text: 'Hi', user: 'Adam' }, { text: 'Hi How are you?', user: 'John' } ];
        const component = shallow(<Messages messages = {messages} />);
          
          expect(component).toMatchSnapshot();
        });
  });