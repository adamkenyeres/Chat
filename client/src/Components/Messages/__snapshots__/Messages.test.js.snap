// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages should render correctly 1`] = `
<ul
  className="Messages-list"
>
  <li
    className="Messages-message"
  >
    <div
      className="Message-content"
    >
      <div
        className="message-user"
      />
      <div
        className="Message-text"
      />
    </div>
  </li>
  <li
    className="Messages-message"
  >
    <div
      className="Message-content"
    >
      <div
        className="message-user"
      />
      <div
        className="Message-text"
      />
    </div>
  </li>
</ul>
`;
